### PR TITLE
Fix multiples in ScienceDirect

### LINKED ISSUE #1510: Sciencedirect: adapt to new journal issue format
https://www.sciencedirect.com/journal/le-pharmacien-hospitalier-et-clinicien/vol/52/issue/4
reported: 
https://forums.zotero.org/discussion/69715/science-direct-translator-broken-on-journal-page#latest

### PR DESCRIPTION
**Pull request by adam3smith**
For new format. Also clean-up some includes. Update tests.

Closes #1510

### CODE REVIEW

**Comment by zuphilip** [comment]
Are the tests in Scaffold for you working? I see an error like this
```
19:07:57 Refetching HTML for PDF link
19:07:58 Translation using ScienceDirect failed: 
         string => Error: error parsing XML
         stack => Zotero.Translate.SandboxManager/this.sandbox.DOMParser/wrappedObj.parseFromString@chrome://zotero/content/xpcom/translation/translate_firefox.js:387:33
         getPDFLink/<@ScienceDirect:101:13
         Zotero.Utilities.Translate.prototype.doGet/xmlhttp<@chrome://zotero/content/xpcom/utilities_translate.js:330:5
         _stateChange@chrome://zotero/content/xpcom/http.js:1015:6
         Zotero.HTTP</this.doGet/xmlhttp.onreadystatechange@chrome://zotero/content/xpcom/http.js:368:4
         
         url => http://www.sciencedirect.com/science/article/pii/S0896627311004430#bib5
         downloadAssociatedFiles => true
         automaticSnapshots => true
```
However, if I load an page in Scaffold Browser then it works fine...

File: ScienceDirect.js [hunk 1]
```diff
@@ -25,23 +25,24 @@ function detectWeb(doc, url) {
...
 
-	if ((url.indexOf("pdf") !== -1 &&
-			url.indexOf("_ob=ArticleURL") === -1 &&
-			url.indexOf("/article/") === -1) ||
+	if ((url.includes("pdf") &&
+			!url.includes("_ob=ArticleURL") &&
+			!url.includes("/article/")) ||
 		url.search(/\/(?:journal|bookseries|book|handbooks|referenceworks)\//) !== -1 ||
-		url.indexOf("_ob=ArticleListURL") !== -1) {
+		url.includes("_ob=ArticleListURL")) {
```
Comment (zuphilip) [inline ScienceDirect.js]: Do you happen to know for what this last case block here is about? We don't have anymore an test url including `_ob=ArticleListURL`.

This was first introduced by the commit https://github.com/zotero/translators/commit/06047f8893d04b703ae0d692a1906acf2df793c1

**Comment by zuphilip** [comment]
This looks fine to fix the issue with multiples in the issue pages. Did you consider also the "Latest published articles" or "Articles in Press" on the journal page? E.g. http://www.sciencedirect.com/journal/academic-pediatrics/ ? I don't know if this could be done easily and in a reliable way.

(BTW there are just a handful more `.indexOf` occurrences in the translator which you could replace.)

**Comment by adam3smith** [comment]
I have the same failure in Scaffold, yes -- I tried to play around with this a little (switching the doGet to a processDocuments, as @dstillman's TODO comment suggested, but couldn't get that to work reliably and as for me, too, this works both in Scaffold and in Zotero I figured we can postpone that.

I did look at the journal pages like  http://www.sciencedirect.com/journal/academic-pediatrics/  but decided that it's not necessarily user expectation that Zotero would work there and was worried about false positives on item pages. 

Removed the remaining instances of indexOf and the ArticleListURL, which I agree does no longer appear to exit

**Comment by zuphilip** [comment]
Okay. How did you then updating the tests?

File: ScienceDirect.js [hunk 2]
```diff
@@ -18,18 +18,17 @@ function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelec
...
 		doc.title == "ScienceDirect - Dummy" ||
-		(url.indexOf("/science/advertisement/") !== -1)) {
+		(url.includes("/science/advertisement/"))) {
 		return false;
 	}
 
 	if ((url.includes("pdf") &&
 			!url.includes("_ob=ArticleURL") &&
 			!url.includes("/article/")) ||
-		url.search(/\/(?:journal|bookseries|book|handbooks|referenceworks)\//) !== -1 ||
```
Comment (zuphilip) [inline ScienceDirect.js]: The ending parenthesis is missing here. I am astonished that this is not a test breaking issue. Let me check...
```
$ ./.ci/checkTranslator.sh ScienceDirect.js
# WARN:  ScienceDirect.js :Must be AGPL licensed
# WARN:  ScienceDirect.js :JSHint shows issues with this code
# WARN:  ScienceDirect.js :stdin:16:184: Missing semicolon.
# WARN:  ScienceDirect.js :stdin:16:356: Missing semicolon.
# WARN:  ScienceDirect.js :stdin:31:87: Expected ')' to match '(' from line 28 and instead saw '{'.
# WARN:  ScienceDirect.js :stdin:55:5: Unrecoverable syntax error. (10% scanned).
# WARN:  ScienceDirect.js :
# WARN:  ScienceDirect.js :4 errors
ok - ScienceDirect.js (Warnings: badLicense problematicJS)
```
I will fix these small nits...

**Comment by zuphilip** [comment]
Thank you @adam3smith !

**Pull request merged**